gobblin-data-management/src/main/java/org/apache/gobblin/data/management/trash/Trash.java

```diff
@@ -94,6 +97,8 @@ public Path getTrashLocation() throws IOException {
    */
   protected Path createTrashLocation(FileSystem fs, Properties props, String user) throws IOException {
     Path trashLocation;
+    new DestinationDatasetHandlerService(new State(props), false, null)
```

Comment:
What will this do? Add something to props? Can we get the trash folder or dataset root folder directly, through the statically-typed APIs, instead of a global collection of props?

Reply:
Yes, is supposed to add configs to props. I thought about having a direct method to determine trash location, but we then will need to add a method in interface DestinationDatasetHandler. Should i do it that way?
Another way is to create a custom trash factory.

Reply:
You can take a look on how this can be refactored better, to make the logic more obvious.